### PR TITLE
updated about-protected-branches.md with new Liquid template helper

### LINKED ISSUE #24498: We list an incorrect branch protection rule
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/enterprise-server@3.5/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-deployments-to-succeed-before-merging

### What part(s) of the article would you like to see updated?

The list of servers for this rule are incorrect. The specific rule: `Require deployments to succeed before merging` made it into GHES versions 3.6 and up, but we list about all GA versions of GHES including versions that do not have this rule.

### Additional information

[Conversation when asking about this](https://github.slack.com/archives/C02CN6WKQUW/p1678992698925959)
[ZenDesk ticket where I was initially asked](https://github.zendesk.com/agent/tickets/2061773)

### PR DESCRIPTION
**Pull request by PaoloJr**
### Why:
request to update documentation with filtering by product-type for the section `Require deployments to succeed before merging`

Closes: #24498

### What's being changed (if available, include any code snippets, screenshots, or gifs):
a Liquid template helper has been added to the section `Require deployments to succeed before merging` in order to filter which product (Free, Pro, Team, GHES, GHEA etc.) has this feature

### Check off the following:

- [X] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by cmwilson21** [comment]
@PaoloJr90 Thanks for submitting another PR! 🎉 

I'll get this triaged for review! :zap:

**Review by sophietheking — DISMISSED** [review]
👋 @PaoloJr90 Thanks for opening this PR and contributing to our Docs! 💖 I only have one comment about the period you added, which renders like this on staging:

period-review

File: content/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Secret gists don't show up in {% data variables.gists.discover_url %} and are no
 
 If your site administrator has disabled private mode, you can also use anonymous gists, which can be public or secret.
 
-{% data reusables.gist.anonymous-gists-cannot-be-deleted %}
+{% data reusables.gist.anonymous-gists-cannot-be-deleted %}.
```
[the remark this answers is not in the document]
Comment (PaoloJr90) [reply content/get-started/writing-on-github/editing-and-sharing-content-with-gists/creating-gists.md:32]: thanks for carefully reviewing. It appears that I did not check the Prettier added a period upon saving 👍

**Review by sophietheking — APPROVED** [review]

**Comment by sultanabdalmohsen** [comment]
> 

> @PaoloJr90شكرا لتقديم PR آخر!🎉
> 
> سأحضر هذا الفرز للمراجعة! ⚡
> 
> نصيحة سريعة - إذا قمت بإزالة السطر الإضافي الفاصل بين `Closes:`المشكلة التي تقوم بإغلاقها ، فسيتم تلقائيًا ربط المشكلة والعلاقات العامة معًا للحصول على مراجعة أسرع وإغلاق الإصدار التلقائي عند دمج العلاقات العامة. لقد قمت بتحديثه لهذا ، ولكن مجرد تنبيه للعلاقات العامة المستقبلية

**Pull request merged**